### ISSUE TITLE
Loss values become "nan" while trining a LSTM model

### ISSUE THREAD

**Issue by GeorgeShimakawa** [plaidml/plaidml#220]
When I changed Keras backend from cpu to an AMD GPU "PlaidML" by adding a line :
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
The loss values became "nan" while running a music composition python code using Istm.

Platform is iMac Late2015, OSX-Mojave, Python 3.6.5. Memory is 16GB.
OpenCL version is 1.2 and the gpu is AMD Radeon R9 M390. CPU is Intel Core i5.
PlaidML version is 0.3.5. 

The output after parsing about 60 midi files, first loss was indicated as 'nan' as follows:
.
.
Parsing midi_songs/sonat-5.mid
Parsing midi_songs/352.mid
Parsing midi_songs/384.mid
Parsing midi_songs/5112.mid
Parsing midi_songs/sonat-4.mid
INFO:plaidml:Opening device "metal_amd_radeon_r9_m390.0"
Epoch 1/200
INFO:plaidml:Analyzing Ops: 2104 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 5776 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 9603 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 14197 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 19025 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 24197 of 29742 operations complete
INFO:plaidml:Analyzing Ops: 27148 of 29742 operations complete
720/43019 [..............................] - ETA: 1:39:10 - loss: nan

Even when I continued further the the loss continues to be “nan”.
The loss values were not “nan” when cpu version of Keras was used.
Could you guess any possible cause ?

**Comment by leedrake5** [comment]
I don't know if this helps at all, but I had a similar problem. When I switched from OpenCL to Metal, things got a lot better.

**Comment by GeorgeShimakawa** [comment]
Thanks for your quick response. I switched to Metal by re-setting up PlaidML and I got same results. When I looked around the web, Metal seems not supporting the GPU in my iMac (Radeon M390).

**Comment by brianretford** [comment]
Is there a reproduction case you can point us at? Or can you verify with the latest version 0.6.0? This is usually caused by bugs in our backend implementation. We'll be providing instructions on how to run our backend tests which will help us debug issues like these.

**Comment by dgkutnic** [comment]
Issues in the `needs info` state for 7+ days will be marked as obsolete and closed. If the issue is still occurring for you, please re-open the issue and provide us with the requested info.
